bcipy/simulator/task/copy_phrase.py

```diff
@@ -0,0 +1,138 @@
+    """Get the evidence type provided by the given model"""
+    return model.metadata.evidence_type or DEFAULT_EVIDENCE_TYPE
+
+
+class SimulatorCopyPhraseTask(RSVPCopyPhraseTask):
+    """CopyPhraseTask that simulates user interactions by sampling data
+    from a DataSampler."""
+
+    def __init__(self, parameters: Parameters, file_save: str,
+                 signal_models: List[SignalModel],
+                 language_model: LanguageModel, samplers: Dict[SignalModel,
+                                                               Sampler]):
+        super().__init__(win=None,
+                         daq=None,
+                         parameters=parameters,
+                         file_save=file_save,
+                         signal_models=signal_models,
+                         language_model=language_model,
+                         fake=False)
+        self.save_session_every_inquiry = False
+        self.samplers = samplers
+
+    def init_evidence_evaluators(
+            self, signal_models: List[SignalModel]) -> List[EvidenceEvaluator]:
+        # Evidence will be sampled so we don't need to evaluate raw data.
+        return []
+
+    def init_evidence_types(
+            self, signal_models: List[SignalModel],
+            evidence_evaluators: List[EvidenceEvaluator]
+    ) -> List[EvidenceType]:
+        evidence_types = set(
+            [get_evidence_type(model) for model in self.signal_models])
+        return [EvidenceType.LM, *evidence_types]
+
+    def init_display(self) -> Display:
+        return NullDisplay()
+
+    def init_feedback(self) -> Optional[VisualFeedback]:
+        return None
+
+    def user_wants_to_continue(self) -> bool:
+        return True
+
+    def wait(self, seconds: Optional[float] = None) -> None:
+        """Override to do nothing"""
+
+    def present_inquiry(
+        self, inquiry_schedule: InquirySchedule
+    ) -> Tuple[List[Tuple[str, float]], bool]:
+        """Override ; returns empty timing info; always proceed for inquiry
+        preview"""
+        return [], True
+
+    def show_feedback(self, selection: str, correct: bool = True) -> None:
+        """Override to do nothing"""
+        # TODO: consider tracking metrics
+
+    def compute_button_press_evidence(
+            self, proceed: bool) -> Optional[Tuple[EvidenceType, List[float]]]:
+        # TODO: consider adding a sampler for this
+        return None
+
+    def compute_device_evidence(
+            self,
+            stim_times: List[List],
+            proceed: bool = True) -> List[Tuple[EvidenceType, List[float]]]:
+
+        current_state = self.get_sim_state()
+        self.logger.info("Computing evidence with sim_state:")
+        self.logger.info(current_state)
+
+        evidences = []
+
+        for model in self.signal_models:
+            sampler = self.samplers[model]
+            # This assumes that sampling is independent. Changes to the sampler API are needed if
```

Comment:
Does that mean we assume a uniform LM throughout the simulation?

Reply:
The language model evidence gets added in a different method (in the copy phrase wrapper `initialize_series`). This method specifically gets evidence from devices. So any language model can be configured for simulation.